src/main/java/org/candlepin/subscriptions/tally/MetricUsageCollector.java

```diff
@@ -388,24 +389,29 @@ private void updateUsage(AccountUsageCalculation calc, Event event) {
               getHardwareMeasurementType(
                   getHostHardwareType(event.getHardwareType()),
                   getCloudProviderAsString(event.getCloudProvider()));
-          event
-              .getMeasurements()
+          event.getMeasurements().stream()
               .forEach(
                   measurement -> {
-                    var metricId =
-                        MetricId.fromString(
-                            Optional.ofNullable(measurement.getMetricId())
-                                .orElse(measurement.getUom()));
-                    if (Variant.getMetricsForTag(productId).stream()
-                        .map(Metric::getId)
-                        .anyMatch(definedMetricId -> definedMetricId.equals(metricId.toString()))) {
+                    var metricId = toMetricId(measurement);
+                    if (isMetricSupportedByProduct(metricId, productId)) {
```

Comment:
I think we should log a warning when we skip a metric.

Reply:
Added